core/corehttp/gateway_handler.go

```diff
@@ -334,45 +325,55 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	// set these headers _after_ the error, for we may just not have it
-	// and don't want the client to cache a 500 response...
-	// and only if it's /ipfs!
-	// TODO: break this out when we split /ipfs /ipns routes.
-	modtime := time.Now()
-
-	if f, ok := dr.(files.File); ok {
-		if strings.HasPrefix(urlPath, ipfsPathPrefix) {
-			w.Header().Set("Cache-Control", "public, max-age=29030400, immutable")
-
-			// set modtime to a really long time ago, since files are immutable and should stay cached
-			modtime = time.Unix(1, 0)
+	// Support custom response formats passed via ?format or Accept HTTP header
+	if contentType := getExplicitContentType(r); contentType != "" {
+		switch contentType {
+		case "application/vnd.ipld.raw":
+			logger.Debugw("serving raw block", "path", parsedPath)
+			i.serveRawBlock(w, r, resolvedPath.Cid(), parsedPath)
+			return
+		case "application/vnd.ipld.car":
+			logger.Debugw("serving car stream", "path", parsedPath)
+			i.serveCar(w, r, resolvedPath.Cid(), parsedPath)
+			return
+		case "application/vnd.ipld.car; version=1":
+			logger.Debugw("serving car stream", "path", parsedPath)
+			i.serveCar(w, r, resolvedPath.Cid(), parsedPath)
+			return
+		case "application/vnd.ipld.car; version=2": // no CARv2 in go-ipfs atm
```

Comment:
I think this is a too simple matching thing.

I think we should support a more general `application/vnd.ipld.car[ \t]*;[ \t]*version[ \t]*=[ \t]*1` for a better developer experience:
![Capture d’écran du 2022-03-08 21-45-59](https://user-images.githubusercontent.com/24391983/157322101-028fca53-7fda-43c7-9d63-44e2fbec49c0.png)

(note the `[ \t]*` arround the `=` that might seems unneeded, however arround the `;` that seems important to me).
(also if they were `[ \t]?` (optional instead of optional or repeated) I would be fine too)

(also maybe there is a spec somewhere that standardize the white-space placement, in that case ignore this)